### PR TITLE
Additional Umbraco processors

### PR DESCRIPTION
**Pull request by karltynan**
Not just Umbraco, but string manipulation also.

### CODE REVIEW

File: Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		public override object ProcessValue()
+		{
+			if (Value == null) throw new NullReferenceException();
+
+			var stringValue = Value as string;
+
+			// byte array representation of that string
+			byte[] encodedValue = new UTF8Encoding().GetBytes(stringValue);
+
+			// need MD5 to calculate the hash
```
Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs]: This comment is unnecessary

File: Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+	public class StringToMd5HashAttribute : TestableDittoProcessorAttribute
+	{
+		public override object ProcessValue()
+		{
+			if (Value == null) throw new NullReferenceException();
+
+			var stringValue = Value as string;
+
+			// byte array representation of that string
```
Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs]: This comment doesn't mean any more than the line it's commenting please remove

File: Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+			byte[] encodedValue = new UTF8Encoding().GetBytes(stringValue);
+
+			// need MD5 to calculate the hash
+			byte[] hashValue = ((HashAlgorithm)CryptoConfig.CreateFromName("MD5")).ComputeHash(encodedValue);
+
+			// string representation (similar to UNIX format)
+			return BitConverter.ToString(hashValue).Replace("-", string.Empty).ToLower();
```
Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs]: Why do we replace the dashes?

Comment (karltynan) [inline Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs]: Not sure - found this method online and it worked for converting an email into MD5 hash for Gravatar (which is the use case)

Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs]: I'd think that if a hash contains a dash that would need to stay I don't think md5 hashes would though

Comment (karltynan) [inline Gibe.DittoProcessors/Processors/StringToMd5HashAttribute.cs]: The hash gets returned like so: `07-bd-b0-98-0f-3f-74-13-f0-1a-43-67-d6-bc-61-49`, which is then converted to `07bdb0980f3f7413f01a4367d6bc6149`.

File: Gibe.DittoProcessors/Processors/StringIsNotEmptyAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace Gibe.DittoProcessors.Processors
+{
+	public class StringIsNotEmptyAttribute : TestableDittoProcessorAttribute
```
Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/StringIsNotEmptyAttribute.cs]: Maybe rename StringIsNotNullOrEmptyAttribute

**Pull request merged**